### DISCUSSION TITLE
coqui TTS for web

### DISCUSSION THREAD

**Discussion by jozefchutka** [coqui-ai/TTS#1390] [General]
Hi,

I wonder if there are any options running coqui  TTS on web on client side (javascript/wasm) and if achievable and convenient. The use case is rather simple, (no model training considered), using existing models to provide TTS feature in client side javascript. Lets say convert text to a playable audio format .wav etc.

Are you aware of existing projects, or any ideas how this could be achieved?

Consdering the performance, on an average machine without HW acceleration (like js), how fast coqui TTS is in general? Looking at https://erogol.github.io/ddc-samples/  , say the sentence " Bill got in the habit of asking himself “Is that thought true?” and if he wasn’t absolutely certain it was he just let it go." is that rather instant or takes some considerable time to process?

**Comment by guest271314** [comment]
If you are able to download and build TTS you can communicate with the application using Native Messaging.

**Comment by jozefchutka** [comment]
The idea is to run whole coqui TTS on client side in browser. Web platform, including webassembly, is mature enough to handle complex libs. I am curios if someone already tried... anyone?

**Comment by guest271314** [comment]
How do you propose to handle with the 700MB+ PyTorch dependency?

**Reply by Mika-IO** [reply]
Download it and store it in the browser cache. A similar example is https://webllm.mlc.ai/, but I believe that in-browser TTS models would be more helpful and useful than LLMs.

**Reply by guest271314** [reply]
rhasspy/piper doesn't depend on a 700 MB PyTorch https://github.com/guest271314/native-messaging-piper

**Comment by Acumane** [comment]
Had the same thought. I might just read from the system clipboard instead

**Reply by guest271314** [reply]
The last time I checked the PyTorch 700MB dependency was prohibitive on the temporary file system running on RAM I tried to build on. 

Have you tried compiling to WASM?
